addons/mail/static/src/models/thread_viewer/thread_viewer_container.js

```diff
@@ -0,0 +1,129 @@
+
+function factory(dependencies) {
+
+    class ThreadViewerContainer extends dependencies['mail.model'] {
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @param {mail.thread_cache} threadCache
+         * @param {string} scrollTop
+         */
+        saveThreadCacheScrollPositionsAsInitial(threadCache, scrollTop) {
+            this.update({
+                threadCacheInitialScrollPositions: Object.assign({}, this.threadCacheInitialScrollPositions, {
+                    [threadCache.localId]: scrollTop,
+                }),
+            });
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computeStringifiedDomain() {
+            if (this.discuss) {
+                return this.discuss.stringifiedDomain;
+            }
+            return '[]';
+        }
+
+        /**
+         * @private
+         * @returns {mail.thread|undefined}
+         */
+         _computeThread() {
+            if (this.chatter) {
+                return [['link', this.chatter.thread]];
+            }
+            if (this.chatWindow) {
+                return [['link', this.chatWindow.thread]];
+            }
+            if (this.discuss) {
+                return [['link', this.discuss.thread]];
+            }
+            return [['unlink']];
+        }
+
+        /**
+         * @private
+         * @returns {mail.thread_viewer|undefined}
+         */
+        _computeThreadViewer() {
+            if (!this.hasThreadViewer) {
+                return [['unlink']];
+            }
+            if (this.threadViewer) {
+                return [];
+            }
+            return [['create']];
+        }
+
+    }
+
+    ThreadViewerContainer.fields = {
+        /**
+         * Determine whether the current thread viewer should be displayed.
+         *
+         * This field should be controlled by the record which created this
+         * container, and it should otherwise be considered read-only.
+         */
+        hasThreadViewer: attr(),
+        /**
+         * Domain to apply when fetching messages for the current thread.
+         *
+         * This field should be controlled by the record which created this
+         * container, and it should otherwise be considered read-only.
+         */
+        stringifiedDomain: attr({
+            default: '[]',
+        }),
+        /**
+         * Thread currently displayed by this thread viewer.
+         *
+         * This field should be controlled by the record which created this
+         * container, and it should otherwise be considered read-only.
+         */
+        thread: many2one('mail.thread'),
```

Comment:
Should be computed or related fields (source of truth in full-fledge UI record, this model represents a reusable module of a UI record)

Reply:
So I keep/reintroduce the obsolete "computes" still present on this model, with the corresponding inverse relations to discuss, chatter, chatWindow, ... ?